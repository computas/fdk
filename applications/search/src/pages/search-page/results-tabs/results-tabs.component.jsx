import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FeatureToggle } from 'react-feature-toggles';
import cx from 'classnames';

import { addOrReplaceParamWithoutURL } from '../../../lib/addOrReplaceUrlParam';
import localization from '../../../lib/localization';
import './results-tabs.scss';
import { FEATURES } from '../../../app/features';
import {
  PATHNAME_DATASETS,
  PATHNAME_APIS,
  PATHNAME_CONCEPTS
} from '../../../constants/constants';

export const ResultsTabs = props => {
  const {
    activePath,
    searchParam,
    countDatasets,
    countTerms,
    countApis
  } = props;

  const search = addOrReplaceParamWithoutURL(searchParam, 'from', '');
  return (
    <div className="row">
      <div className="col-12 col-lg-8 offset-lg-4">
        <ul className="search-results-tabs d-flex align-items-center">
          <li
            className={cx('d-flex justify-content-center', {
              'li-active': activePath === PATHNAME_DATASETS
            })}
          >
            <Link
              to={{ pathname: PATHNAME_DATASETS, search }}
              aria-label="Link til side for datasett:"
            >
              {localization.page.datasetTab}
              <span>&nbsp;({countDatasets})</span>
            </Link>
          </li>
          <FeatureToggle featureName={FEATURES.API}>
            <li
              className={cx('d-flex justify-content-center beta', {
                'li-active': activePath === PATHNAME_APIS
              })}
            >
              <Link
                to={{ pathname: PATHNAME_APIS, search }}
                aria-label="Link til side for api:"
              >
                {localization.page.apiTab}
                <span>&nbsp;({countApis})</span>
              </Link>
            </li>
          </FeatureToggle>
          <li
            className={cx('d-flex justify-content-center beta', {
              'li-active': activePath === PATHNAME_CONCEPTS
            })}
          >
            <Link
              to={{ pathname: PATHNAME_CONCEPTS, search }}
              aria-label="Link til side for begrep:"
            >
              {localization.page.termTab}
              <span>&nbsp;({countTerms})</span>
            </Link>
          </li>
        </ul>
      </div>
    </div>
  );
};

ResultsTabs.defaultProps = {
  activePath: null,
  searchParam: '',
  countDatasets: null,
  countTerms: null,
  countApis: null
};

ResultsTabs.propTypes = {
  activePath: PropTypes.string,
  searchParam: PropTypes.string,
  countDatasets: PropTypes.number,
  countTerms: PropTypes.number,
  countApis: PropTypes.number
};
