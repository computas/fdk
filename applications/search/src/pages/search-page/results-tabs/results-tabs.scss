@import '../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.search-results-tabs {
  margin-bottom: 0;
  padding-left: 0;
  li {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    border-color: $color3;
    list-style-type: none;
    button, a {
      background: none;
      border: 0;
      box-sizing: border-box;
      color: $color1;
      padding: .8em 1em;
      font-weight: 500;
      text-decoration: none;
    }
  }

  .li-active {
    background-color: $color6;
    font-size: 1.8rem;
    &.beta {
      background-image: url("/static/img/banner-beta-active.svg");
      background-repeat: no-repeat;
      background-position: right;
      background-size: 3em;
    }
  }
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {
  .search-results-tabs {
    li {
      line-height: 1em;
      min-width: 10em;
      button, a {
        padding: 1em 2em;
      }
    }

    .li-active {
      font-size: 1.8rem;
    }
  }
}