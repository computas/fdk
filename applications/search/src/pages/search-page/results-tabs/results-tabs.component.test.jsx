import React from 'react';
import { shallow } from 'enzyme';
import { ResultsTabs } from './results-tabs.component';
import {
  PATHNAME_DATASETS,
  PATHNAME_CONCEPTS
} from '../../../constants/constants';
import defaultProps from './__fixtures/defaultProps.json';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ResultsTabs {...defaultProps} />);
});

test('should render ResultsTabs with "Dataset" active correctly', () => {
  wrapper.setProps({
    activePath: PATHNAME_DATASETS
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ResultsTabs with "Concepts" active correctly', () => {
  wrapper.setProps({
    activePath: PATHNAME_CONCEPTS
  });
  expect(wrapper).toMatchSnapshot();
});
