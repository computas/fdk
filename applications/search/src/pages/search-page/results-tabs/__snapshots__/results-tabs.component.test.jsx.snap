// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsTabs with "Concepts" active correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-lg-8 offset-lg-4"
  >
    <ul
      className="search-results-tabs d-flex align-items-center"
    >
      <li
        className="d-flex justify-content-center"
      >
        <Link
          aria-label="Link til side for datasett:"
          replace={false}
          to={
            Object {
              "pathname": "/",
              "search": "?",
            }
          }
        >
          Dataset
          <span>
             (
            985
            )
          </span>
        </Link>
      </li>
      <FeatureToggle
        featureName="API"
        showOnlyWhenDisabled={false}
      >
        <li
          className="d-flex justify-content-center beta"
        >
          <Link
            aria-label="Link til side for api:"
            replace={false}
            to={
              Object {
                "pathname": "/apis",
                "search": "?",
              }
            }
          >
            Api
            <span>
               (
              4
              )
            </span>
          </Link>
        </li>
      </FeatureToggle>
      <li
        className="d-flex justify-content-center beta li-active"
      >
        <Link
          aria-label="Link til side for begrep:"
          replace={false}
          to={
            Object {
              "pathname": "/concepts",
              "search": "?",
            }
          }
        >
          Concepts
          <span>
             (
            1
            )
          </span>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;

exports[`should render ResultsTabs with "Dataset" active correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-lg-8 offset-lg-4"
  >
    <ul
      className="search-results-tabs d-flex align-items-center"
    >
      <li
        className="d-flex justify-content-center li-active"
      >
        <Link
          aria-label="Link til side for datasett:"
          replace={false}
          to={
            Object {
              "pathname": "/",
              "search": "?",
            }
          }
        >
          Dataset
          <span>
             (
            985
            )
          </span>
        </Link>
      </li>
      <FeatureToggle
        featureName="API"
        showOnlyWhenDisabled={false}
      >
        <li
          className="d-flex justify-content-center beta"
        >
          <Link
            aria-label="Link til side for api:"
            replace={false}
            to={
              Object {
                "pathname": "/apis",
                "search": "?",
              }
            }
          >
            Api
            <span>
               (
              4
              )
            </span>
          </Link>
        </li>
      </FeatureToggle>
      <li
        className="d-flex justify-content-center beta"
      >
        <Link
          aria-label="Link til side for begrep:"
          replace={false}
          to={
            Object {
              "pathname": "/concepts",
              "search": "?",
            }
          }
        >
          Concepts
          <span>
             (
            1
            )
          </span>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
